client/services/SolutionBrokerService.js

```diff
@@ -18,20 +18,38 @@
  */
 
 tie.factory('SolutionBrokerService', [
```

Comment:
I had some difficulty coming up with a name for this; let's discuss offline.

The main reason I called it that was because this service "handles" an incoming solution, and sends it through a chain of getting feedback from running the code, correctness checks, etc. -- so this is an overarching, "organizational" service. I'm not entirely sure what to name it, though; perhaps we could go with SolutionHandlerService or something. But we can discuss this offline.